### PR TITLE
UCP105 :refactor the class components to function components for src/layout/signin

### LINKED ISSUE #105: UCP: Refactor the class components to function components
## Description

In the frontend react code, we mix the class component and function component together, so this task is to refactor the class components to function components.

100 points for each:

- [x] `packages/dashboard_components/*`
- [x] `packages/keyvis/`
- [x] `src/apps/dashboardSettings/`
- [x] `src/apps/userProfile/`
- [x] `src/components/*`
- [x] `src/layout/signin`

## Score 

* 100

## Mentor(s)

* @baurine  

Contact the mentors: **#tidb-challenge-program** channel in [TiDB Community](https://join.slack.com/t/tidbcommunity/shared_invite/enQtNzc0MzI4ODExMDc4LWYwYmIzMjZkYzJiNDUxMmZlN2FiMGJkZjAyMzQ5NGU0NGY0NzI3NTYwMjAyNGQ1N2I2ZjAxNzc1OGUwYWM0NzE)  Slack Workspace

## Recommended Skills

* React

### PR DESCRIPTION
**Pull request by gauss1314**
UCP #105

This close #105 issue partly.

## What problem does this PR solve?
Refactor the class components to function components for `src/layout/signin`.

## What are the type of the changes?
Improvement

### CODE REVIEW

File: ui/src/layout/signin/RootComponent.js [hunk 1]
```diff
@@ -27,22 +27,16 @@ const AnimationItem = (props) => {
   )
 }
 
-@withTranslation()
-class TiDBSignInForm extends React.PureComponent {
-  state = {
-    loading: false,
-    signInError: null,
-  }
+function TiDBSignInForm({ t, registry }) {
```
Comment (baurine) [inline ui/src/layout/signin/RootComponent.js:30]: Maybe you can directly use the `t` by `const { t } = useTranslation()` inside the component, so you don't need to pass it via props by `withTranslation()(App)`, you can have a try.

Comment (gauss1314) [reply ui/src/layout/signin/RootComponent.js]: Done.

File: ui/src/layout/signin/RootComponent.js [hunk 2]
```diff
@@ -27,22 +27,17 @@ const AnimationItem = (props) => {
...
+  const [loading, setLoading] = useState(false)
+  const [signInError, setSignInError] = useState(null)
 
-  constructor(props) {
-    super(props)
-    this.refForm = React.createRef()
-    this.refPassword = React.createRef()
-  }
+  const refForm = React.createRef()
+  const refPassword = React.createRef()
```
Comment (baurine) [inline ui/src/layout/signin/RootComponent.js:36]: Does `createRef()` work inside the function componest, I remember it only wors inside the class component, see this: https://github.com/dt-fe/weekly/blob/v2/141.%E7%B2%BE%E8%AF%BB%E3%80%8AuseRef%20%E4%B8%8E%20createRef%20%E7%9A%84%E5%8C%BA%E5%88%AB%E3%80%8B.md

Maybe we need to use `useRef()`.

Comment (gauss1314) [reply ui/src/layout/signin/RootComponent.js]: Thanks! I use `useRef()` now, PTAL.

Comment (baurine) [reply ui/src/layout/signin/RootComponent.js]: Nice, the rest LGTM, please help merge the latest master branch code.

**Comment by gauss1314** [comment]
> LGTM

@baurine Done. One more question , I should get 100 points for each PR, right? But when last PR merged, nothing happened~

**Comment by baurine** [comment]
> > LGTM
> 
> @baurine Done. One more question , I should get 100 points for each PR, right? But when last PR merged, nothing happened~

Yep, it should do. @you06 can you help have a look.

File: ui/src/layout/signin/RootComponent.js [hunk 3]
```diff
@@ -51,153 +46,144 @@ class TiDBSignInForm extends React.PureComponent {
...
+  const clearErrorMessages = () => {
+    setSignInError(null)
   }
 
-  componentDidMount = () => {
-    this.refPassword.current.focus()
-  }
+  useEffect(() => {
+    refPassword.current.focus()
+  })
```
Comment (breeswish) [inline ui/src/layout/signin/RootComponent.js:81]: should be `useEffect(() => ..., [])` instead?

**Comment by you06** [comment]
@gauss1314 the points should be fixed.

File: ui/src/layout/signin/RootComponent.js (hunk 3 above)
Comment (baurine) [reply ui/src/layout/signin/RootComponent.js:81]: Yep, @gauss1314 , can you fix this?

Comment (gauss1314) [reply ui/src/layout/signin/RootComponent.js:81]: Oh, by passing an empty array, only execute the side effect once (at mount time). Is that right? I will raise a PR to fix it.

**Comment by breeswish** [comment]
@all-contributors please add @gauss1314 for code

**Pull request merged**